terraform/main.tf

```diff
@@ -0,0 +1,64 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+    }
+  }
+}
+
+locals {
+  cred = file("cred.json")
+  email = lookup(jsondecode(local.cred), "client_email")
+}
+
+provider "google" {
+  credentials = local.cred
+  project = var.project_id
+  region  = var.region
+  zone    = "us-central1-c"
+}
+
+# Define and deploy a workflow
+resource "google_workflows_workflow" "test_runner_workflow" {
+  name = "test-runner-workflow"
+  region = var.region
+  description = "Test runner workflow"
+  service_account = local.email
+  # Import workflow YAML file
+  source_contents = file("workflow.yaml")
+
+  depends_on = [google_project_service.workflows]
+}
+
+# Define and deploy a tasks queue
+resource "google_cloud_tasks_queue" "test_runner_tasks_queue" {
+  name = "test-runner-tasks-queue"
+  location = var.region
+
+  rate_limits {
+    max_concurrent_dispatches = 1000
```

Comment:
does this rate need to be so high? can it be something like 3? Maybe even 1?

Reply:
yes, I set it to 1.